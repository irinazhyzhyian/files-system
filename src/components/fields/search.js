import React, { useState } from "react";
import { Input, Button } from '@mui/joy';


export const Search = ({ onSearch }) => {
  const [searchText, setSearchText] = useState('');

  const handleSearchTextChange = (event) => {
    const { value } = event.target;

    setSearchText(value);
  }

  const handleSearch = () => {
    onSearch(searchText);
  }

  return (
    <Input
      placeholder="Search..."
      value={searchText}
      sx={{width: 400}}
      onChange={handleSearchTextChange}
      id="searchField"
      endDecorator={
        <Button
          variant="solid"
          color="primary"
          type="submit"
          disabled={!searchText || searchText.length < 3}
          onClick={handleSearch}
        >
          Find
        </Button>
      }
    />
  );
} 