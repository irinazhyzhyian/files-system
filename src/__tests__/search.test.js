import { Search } from "../components/fields/search";
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Search component', () => {
  it('snapshot', () => {
    const component = renderer.create(
      <Search />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('render search button', async () => {
    const onSearchMock = jest.fn();
    render(
      <Search onSearch={onSearchMock} />
    );
  
    const inputElement = screen.getByPlaceholderText("Search...");
    const findButtonElement = screen.getByText("Find");
  
    fireEvent.change(inputElement, { target: { value: "test" } });
  
    fireEvent.click(findButtonElement);
  
    expect(onSearchMock).toHaveBeenCalledWith("test");
  })

  it("search function is not called when input is empty", () => {
    const onSearchMock = jest.fn();
    render(<Search onSearch={onSearchMock} />);
  
    const findButtonElement = screen.getByText("Find");
  
    fireEvent.click(findButtonElement);
  
    expect(onSearchMock).not.toHaveBeenCalled();
  });
});