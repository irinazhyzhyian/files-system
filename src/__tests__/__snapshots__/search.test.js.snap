// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapshot 1`] = `
<div
  className="MuiInput-root MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeMd css-1ytylkz-JoyInput-root"
  onClick={[Function]}
>
  <input
    className="MuiInput-input css-1moyr10-JoyInput-input"
    disabled={false}
    id="searchField"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search..."
    required={false}
    value=""
  />
  <div
    className="MuiInput-endDecorator css-ki6l96-JoyInput-endDecorator"
  >
    <button
      className="MuiButton-root Mui-disabled MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cua13l-JoyButton-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      Find
    </button>
  </div>
</div>
`;
